phpunit/class-block-templates-test.php

```diff
@@ -74,6 +74,30 @@ function test_gutenberg_build_template_result_from_post() {
 		$this->assertEquals( 'wp_template', $template->type );
 	}
 
+	function test_inject_theme_attribute_in_content() {
+		$theme                           = get_stylesheet();
+		$content_without_theme_attribute = '<!-- wp:template-part {"slug":"header","align":"full", "tagName":"header","className":"site-header"} /-->';
+		$template_content                = _inject_theme_attribute_in_content(
+			$content_without_theme_attribute,
+			$theme
+		);
+		$this->assertStringContainsString( $theme, $template_content );
```

Comment:
Maybe we should be more explicit? using `parse_blocks( $template_content )` then assert `$blocks[0]['attrs']['theme'] === $theme`? Or contains `'"theme":"$theme"'`.

This can save us a roundtrip if for some reason this function injects the wrong attribute.

On the other hand, we might want to add a test here that checks nothing else is changed other than a `theme` attribute is added.

Reply:
> Maybe we should be more explicit? using parse_blocks( $template_content ) then assert $blocks[0]['attrs']['theme'] === $theme? Or contains '"theme":"$theme"'.

I thought about this. I was wondering if, at that point, we were inadvertently testing implementation details of the method. What if I just compared the entire serialized block we expect with `$template_content` instead?

```php
		$content_without_theme_attribute = '<!-- wp:template-part {"slug":"header","align":"full", "tagName":"header","className":"site-header"} /-->';
		$template_content                = _inject_theme_attribute_in_content(
			$content_without_theme_attribute,
			$theme
		);
		$actual                          = '<!-- wp:template-part {"slug":"header","theme":"$theme","align":"full", "tagName":"header","className":"site-header"} /-->';
		$this->assertStringContainsString( $actual, $template_content );
```

> On the other hand, we might want to add a test here that checks nothing else is changed other than a theme attribute is added.

Yeah. I think if we want to ensure that the wrong attribute isn't injected, this is the way to go.

Reply:
I'm not sure about how attributes are ordered when blocks are converted to markup. If it's consistent all the time (ie. theme is always at the same position), then we can just compare strings as you suggested.

Reply:
> If it's consistent all the time (ie. theme is always at the same position), then we can just compare strings as you suggested.

I'd bet that they are, especially because, by default, associative arrays in PHP are ordered in nature. When we inject the theme attribute, it should always be appended last in the list of attributes (unless the `serialize_blocks` helper method does something wonky)

Reply:
Added the changes